ruby/lib/minitest/queue/runner.rb

```diff
@@ -70,14 +70,50 @@ def run_command
         trap('INT') { Minitest.queue.shutdown! }
 
         if queue.rescue_connection_errors { queue.exhausted? }
-          puts green("All tests were ran already")
+          puts green('All tests were ran already')
         else
           load_tests
           populate_queue
         end
         # Let minitest's at_exit hook trigger
       end
 
+      def grind_command
+        invalid_usage!('No list to grind provided') if grind_list.nil?
+        invalid_usage!('No grind count provided') if grind_count.nil?
+
+        set_load_path
+
+        queue_config.build_id = queue_config.build_id + '-grind'
+
+        reporter_queue = CI::Queue::Redis::Grind.new(queue_url, queue_config)
+        @queue = CI::Queue::Static.new([], queue_config)
+
+        Minitest.queue = queue
+        reporters = [
+          GrindRecorder.new(build: reporter_queue.build)
+        ]
+        if queue_config.statsd_endpoint
+          reporters << Minitest::Reporters::StatsdReporter.new(statsd_endpoint: queue_config.statsd_endpoint)
+        end
+        Minitest.queue_reporters = reporters
+
+        trap('TERM') { Minitest.queue.shutdown! }
+        trap('INT') { Minitest.queue.shutdown! }
+
+        load_tests
+        tests_to_run = File.readlines(grind_list).map(&:strip!).to_set
```

Comment:
Why can't `Queue::File` be used here? It seems to be doing pretty much the same thing.

Worst case I'd rather see some subclass of it if really necessary.